src/chainstate/stacks/boot/costs.clar

```diff
@@ -0,0 +1,495 @@
+
+(define-read-only (cost_analysis_fetch_contract_entry (n uint))
+    {
+        runtime: (linear n u1 u1),
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: (linear n u1 u1)
+    })
+
+(define-read-only (cost_lookup_variable_depth (n uint))
+    (runtime (linear n u1 u1)))
+
+(define-read-only (cost_lookup_variable_size (n uint))
+    (runtime (linear n u1 u0)))
+
+(define-read-only (cost_lookup_function (n uint))
+    (runtime u1))
+
+(define-read-only (cost_bind_name (n uint))
+    (runtime u1))
+
+(define-read-only (cost_inner_type_check_cost (n uint))
+    (runtime (linear n u1 u1)))
+
+(define-read-only (cost_user_function_application (n uint))
+    (runtime (linear n u1 u1)))
+
+(define-read-only (cost_let (n uint))
+    (runtime (linear n u1 u1)))
+
+(define-read-only (cost_if (n uint))
+    (runtime u1))
+
+(define-read-only (cost_asserts (n uint))
+    (runtime u1))
+
+(define-read-only (cost_map (n uint))
+    (runtime u1))
+
+(define-read-only (cost_filter (n uint))
+    (runtime u1))
+
+(define-read-only (cost_len (n uint))
+    (runtime u1))
+
+(define-read-only (cost_fold (n uint))
+    (runtime u1))
+
+(define-read-only (cost_list_cons (n uint))
+    (runtime (linear n u1 u1)))
+
+(define-read-only (cost_type_parse_step (n uint))
+    (runtime u1))
+
+(define-read-only (cost_data_hash_cost (n uint))
+    (runtime (linear n u1 u1)))
+
+(define-read-only (cost_tuple_get (n uint))
+    (runtime (nlogn n u1 u1)))
+
+(define-read-only (cost_tuple_cons (n uint))
+    (runtime (nlogn n u1 u1)))
+
+(define-read-only (cost_add (n uint))
+    (runtime (linear n u1 u1)))
+
+(define-read-only (cost_sub (n uint))
+    (runtime (linear n u1 u1)))
+
+(define-read-only (cost_mul (n uint))
+    (runtime (linear n u1 u1)))
+
+(define-read-only (cost_div (n uint))
+    (runtime (linear n u1 u1)))
+
+(define-read-only (cost_geq (n uint))
+    (runtime u1))
+
+(define-read-only (cost_leq (n uint))
+    (runtime u1))
+
+(define-read-only (cost_le (n uint))
+    (runtime u1))
+
+(define-read-only (cost_ge  (n uint))
+    (runtime u1))
+
+(define-read-only (cost_int_cast (n uint))
+    (runtime u1))
+
+(define-read-only (cost_mod (n uint))
+    (runtime u1))
+
+(define-read-only (cost_pow (n uint))
+    (runtime u1))
+
+(define-read-only (cost_sqrti (n uint))
+    (runtime u1))
+
+(define-read-only (cost_xor (n uint))
+    (runtime u1))
+
+(define-read-only (cost_not (n uint))
+    (runtime u1))
+
+(define-read-only (cost_eq (n uint))
+    (runtime (linear n u1 u1)))
+
+(define-read-only (cost_begin (n uint))
+    (runtime u1))
+
+(define-read-only (cost_hash160 (n uint))
+    (runtime u1))
+
+(define-read-only (cost_sha256 (n uint))
+    (runtime u1))
+
+(define-read-only (cost_sha512 (n uint))
+    (runtime u1))
+
+(define-read-only (cost_sha512t256 (n uint))
+    (runtime u1))
+
+(define-read-only (cost_keccak256 (n uint))
+    (runtime u1))
+
+(define-read-only (cost_secp256k1recover (n uint))
```

Comment:
Secp256k1 verification and recovery is _O(1)_, but it's probably a "bigger" _O(1)_ than hashing.

Reply:
Yes, I think that is true, but we would be updating these constants through benchmarking anyways, so I don't think it needs to be addressed here.